ddtrace/llmobs/_integrations/crewai.py

```diff
@@ -247,6 +274,117 @@ def _get_agent_tools(self, tools):
             formatted_tools.append(tool_dict)
         return formatted_tools
 
+    def _llmobs_set_tags_flow(self, span, args, kwargs, response):
+        inputs = get_argument_value(args, kwargs, 0, "inputs", optional=True) or {}
+        span._set_ctx_items({NAME: span.name or "CrewAI Flow", INPUT_VALUE: inputs, OUTPUT_VALUE: str(response)})
+        return
+
+    def _llmobs_set_tags_flow_method(self, span, args, kwargs, response):
+        flow_instance = kwargs.pop("_dd.instance", None)
+        initial_flow_state = kwargs.pop("_dd.flow_state", {})
+        input_dict = {
+            "args": [str(arg) for arg in args[2:]],
+            "kwargs": {k: str(v) for k, v in kwargs.items()},
+            "flow_state": initial_flow_state,
+        }
+        span_links = (
+            self._flow_span_to_method_to_span_dict.get(span._parent, {}).get(span.name, {}).get("span_links", [])
+        )
+        if span.name in getattr(flow_instance, "_start_methods", []):
+            span_links.append(
+                {
+                    "span_id": str(span._parent.span_id),
+                    "trace_id": format_trace_id(span.trace_id),
+                    "attributes": {"from": "input", "to": "input"},
+                }
+            )
+
+        if span.name in getattr(flow_instance, "_routers", []):
+            # Routers use their result (not method name) to link to the next method, so do the same for spans
+            span_dict = self._flow_span_to_method_to_span_dict.get(span._parent, {}).setdefault(str(response), {})
+            span_dict.update({"span_id": str(span.span_id), "trace_id": format_trace_id(span.trace_id)})
+
+        span._set_ctx_items(
+            {
+                NAME: span.name or "Flow Method",
+                INPUT_VALUE: input_dict,
+                OUTPUT_VALUE: str(response),
+                SPAN_LINKS: span_links,
+            }
+        )
+        return
+
+    def _llmobs_set_span_link_on_flow(self, flow_span, args, kwargs, flow_instance):
```

Comment:
seeing a lot of access to private `_` members in here, what do you think about making a `llmobs_set_span_link_on_flow` method that wraps this method in a `try`/`except`, and calling `llmobs_set_span_link_on_flow` from the patching file? just to be extra safe (although everything is handled safely with `getattr` anyhow, so up to you!)